scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -71,17 +76,25 @@ def __init__(self, infile=None, schema=None, root_name_override=None, root_attri
 
             self.tree = ET.ElementTree(root)
 
+            self._FILEMAP[infile] = (self.tree, self.root, 0.0)
```

Comment:
Using something like a namedtuple for the elements of _FILEMAP would be more clear here: then each of the 3 elements would have a logical name rather than just being accessible via indices 0, 1 and 2 and requiring the reader to know (for example) that element 2 is the modification time.

Reply:
Will do.